presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -64,6 +64,20 @@ public DictionaryBlock(int positionCount, Block dictionary, int[] ids, boolean d
         this(0, positionCount, dictionary, ids, dictionaryIsCompacted, dictionarySourceId);
     }
 
+    public static Block mask(Block sourceBlock, int[] ids)
```

Comment:
I know. But looking at the method signature it's not that obvious.
Maybe rename  `ids` like `visiblePositions` or `retainedPositions`?